lib/faktory_worker/connection_manager.ex

```diff
@@ -1,28 +1,58 @@
 defmodule FaktoryWorker.ConnectionManager do
   @moduledoc false
 
-  use GenServer
-
   alias FaktoryWorker.Connection
+  alias FaktoryWorker.ConnectionManager
+
+  @connection_errors [
+    :closed,
+    :enotconn,
+    :econnrefused
```

Comment:
These are the only error's I have seen whilst manually testing this. There are loads of potential errors and I'm not sure how much we want to handle here. Full list is here http://erlang.org/doc/man/inet.html#type-posix